source/Preprocessor.cpp

```diff
@@ -417,6 +418,23 @@ Trivia Preprocessor::handleTimescaleDirective(Token directive) {
     return Trivia(TriviaKind::Directive, timescale);
 }
 
+Trivia Preprocessor::handleLineDirective(Token directive) {
+    Token lineNumber = expect(TokenKind::IntegerLiteral);
+    Token fileName = expect(TokenKind::StringLiteral);
+    Token level = expect(TokenKind::IntegerLiteral);
+    const SVInt& levNum = std::get<SVInt>(level.numericValue());
+    if (!(levNum == 0 || levNum == 1 || levNum == 2)) {
```

Comment:
Good point, otherwise you could probably end up with some really weird error messages if you had errors after a malformed `line. Will make both this change and the other minor changes.